pkg/v1/remote/write_test.go

```diff
@@ -1269,3 +1269,30 @@ func TestTag(t *testing.T) {
 		t.Errorf("Validate() = %v", err)
 	}
 }
+
+func BenchmarkWrite(b *testing.B) {
+	// unfortunately the registry _and_ the img have caching behaviour, so we need a new registry
+	// and image every iteration of benchmarking.
+	for i := 0; i < b.N; i++ {
+		// set up the registry
+		s := httptest.NewServer(registry.New())
+		defer s.Close()
+
+		// load the image
+		img, err := random.Image(50*1024*1024, 10)
+		if err != nil {
+			b.Fatalf("random.Image(...): %v", err)
+		}
+
+		tagStr := strings.TrimPrefix(s.URL+"/test/image:tag", "http://")
+		tag, err := name.NewTag(tagStr)
+		if err != nil {
+			b.Fatalf("parsing tag (%s): %v", tagStr, err)
+		}
+
+		err = Write(tag, img)
```

Comment:
We probably want to `b.ResetTimer()` right before this, if that's what we're measuring.